servicereportpkg/repair/plugins/spyre_repair.py

```diff
@@ -57,13 +58,46 @@ def fix_user_mem_conf(self, plugin_obj, user_mem_conf_check):
         else:
             user_mem_conf_check.set_note(Notes.FAIL_TO_FIX)
 
+    def check_if_vfio_udev_rule_is_redundant(self, udev_rule):
+        """Check if VFIO udev rule is Redundant"""
+
+        if udev_rule and udev_rule.startswith('SUBSYSTEM=="vfio"'):
+            splits = udev_rule.split(',')
+            length_splits = len(splits)
+            if length_splits  <= 3:
+                splits = [condition.strip() for condition in splits]
+                has_group = any('GROUP' in part for part in splits)
+                has_mode = any('MODE' in part for part in splits)
+                if ((length_splits==2 and (has_mode or has_group)) or
+                    (length_splits==3 and has_mode and has_group) or
+                    (length_splits==1)):
+                    return True
+        return False
+
     def fix_udev_rules_conf(self, plugin_obj, udev_rules_conf_check):
         """Fix VFIO udev rules"""
 
         for config, val in udev_rules_conf_check.get_config_attributes().items():
             if not val["status"]:
-                append_to_file(udev_rules_conf_check.get_file_path(),
-                               "\n"+config)
+                try:
+                    with open(udev_rules_conf_check.get_file_path(), "r", encoding="utf-8") as file:
+                        lines = file.readlines()
+                    updated_lines = []
+                    for line in lines:
+                        stripped_line = line.strip()
+                        # if a vfio udev rule is redundant incorrect or old rule will be
+                        # removed
+                        if self.check_if_vfio_udev_rule_is_redundant(stripped_line) == False:
+                            updated_lines.append(line)
+                except FileNotFoundError:
+                    self.log.error("File not found : %s", udev_rules_conf_check.get_file_path())
+
+        updated_string = "".join(updated_lines)
+        add_to_file(udev_rules_conf_check.get_file_path(),
+                    updated_string)
+        append_to_file(udev_rules_conf_check.get_file_path(),
+                                "\n"+config)
+
```

Comment:
Why don't we add correct config in the updated_lines and avoid calling append_to_file altogether?